addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/handler/HomematicThingHandler.java

```diff
@@ -0,0 +1,284 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.homematic.handler;
+
+import static org.openhab.binding.homematic.internal.misc.HomematicConstants.*;
+
+import java.io.IOException;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StopMoveType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.link.ItemChannelLinkRegistry;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.homematic.converter.ConverterException;
+import org.openhab.binding.homematic.converter.ConverterFactory;
+import org.openhab.binding.homematic.converter.TypeConverter;
+import org.openhab.binding.homematic.internal.communicator.HomematicGateway;
+import org.openhab.binding.homematic.internal.misc.HomematicClientException;
+import org.openhab.binding.homematic.internal.model.HmDatapoint;
+import org.openhab.binding.homematic.internal.model.HmDatapointConfig;
+import org.openhab.binding.homematic.internal.model.HmDatapointInfo;
+import org.openhab.binding.homematic.internal.model.HmDevice;
+import org.openhab.binding.homematic.type.UidUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HomematicThingHandler} is responsible for handling commands, which are sent to one of the channels.
+ *
+ * @author Gerhard Riegler - Initial contribution
+ */
+public class HomematicThingHandler extends BaseThingHandler {
+    private Logger logger = LoggerFactory.getLogger(HomematicThingHandler.class);
+
+    private static final String[] STATUS_DATAPOINT_NAMES = new String[] { DATAPOINT_NAME_UNREACH,
+            DATAPOINT_NAME_CONFIG_PENDING, DATAPOINT_NAME_DEVICE_IN_BOOTLOADER };
+
+    private ItemChannelLinkRegistry itemChannelLinkRegistry;
+
+    public HomematicThingHandler(Thing thing, ItemChannelLinkRegistry itemChannelLinkRegistry) {
+        super(thing);
+        this.itemChannelLinkRegistry = itemChannelLinkRegistry;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialize() {
+        if (getThing().getChannels().size() == 0) {
+            updateStatus(ThingStatus.INITIALIZING);
```

Comment:
ThingHandlers should not set the INITIALIZING state - this is done by the framework. If everything is alright, you should actually set it to ONLINE.